chain/chain/src/chain.rs

```diff
@@ -1032,8 +1031,7 @@ impl Chain {
         Ok(())
     }
 
-    #[cfg(test)]
-    pub(crate) fn mark_block_as_challenged(
+    pub fn mark_block_as_challenged(
```

Comment:
Is this intentional? Looks like a merge artifact. My general advice for working on PRs would be to avoid merges and to do rebases instead. My usual workflow when not using a fork would be

```console
# check that remote has correct name/url
$ git remote -v
origin	git@github.com:near/nearcore.git (fetch)
origin	git@github.com:near/nearcore.git (push)

# Check that
#   * I am on a feature branch
#   * local master branch exists
#   * local mastre branch tracks remote master ([origin/master] bit)
$ git branch -vv
* feature-branch f6095ed7a test: Snapshot tests for estimator warehouse (#6780)
  master         f6095ed7a [origin/master] test: Snapshot tests for estimator warehouse (#6780)

# Switch to master and update it to match upstraem
$ git switch master && git pull --rebase

# Verify that we are indeed in sync: the top commit is the same for local master and remote origin/master
$ git log
commit f6095ed7a8fb412c1c38bf679a4b60e6593b0f99 (HEAD -> master, origin/master, origin/HEAD, feature-branch)

# Switch back to the feature branch
$ git switch -

# Actually rebase my local feature branch on top of my local master (which is now in sync with remote master)
$ git rebase master
Successfully rebased and updated refs/heads/feature-branch.

# Verify that the rebase went as planned
$ git log
# this is my work on the feature branch (note HEAD -> feature-branch)
commit 2fd220687ad3f6845e8b003e80be0aa8a87616de (HEAD -> feature-branch)
Author: Aleksey Kladov <aleksey.kladov@gmail.com>
Date:   Wed May 11 13:41:33 2022 +0100

    feat: the best ever

# and now it sits atop of master (note origin/master, origin/HEAD, master)
commit f6095ed7a8fb412c1c38bf679a4b60e6593b0f99 (origin/master, origin/HEAD, master)
Author: Jakob Meier <mail@jakobmeier.ch>
Date:   Wed May 11 13:42:51 2022 +0200

# Push rebase to GH
$ git push --force-with-lease
```

Reply:
> Is this intentional? Looks like a merge artifact.

No, sorry, fixed.

> My usual workflow when not using a fork would be

Thanks, this is useful. I just need to find how to make it work with CLion.